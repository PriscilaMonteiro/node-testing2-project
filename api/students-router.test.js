/* eslint-disable no-undef */
// with supertest we always import the whole server
// regardless of what router we are testing here specifically
const request = require('supertest');
const db = require('../data/dbConfig');
const server = require('./server');

test('it is the correct environment for the tests', () => {
    expect(process.env.NODE_ENV).toBe('testing');
});

beforeAll(async () => {
    await db.migrate.rollback();
    await db.migrate.latest();
});

beforeEach(async () => {
    await db.seed.run();
});

afterAll(async () => {
    await db.destroy();
});

describe('[GET] /students', () => {
    it('should return a 200 OK status', async () => {
        const res = await request(server).get('/api/students');
        expect(res.status).toBe(200);
    });
    it('should return JSON', async () => {
        const res = await request(server).get('/api/students');
        expect(res.type).toBe('application/json');
    });
});

describe('[POST] /api/students', () => {
    let res
    beforeEach(async () => {
        res = await request(server)
           .post('/api/students')
           .send({ student_name: 'Alexander' })
    });     
    it('responds with new student', async () => {
        expect(res.body).toMatchObject({ student_id: 5, student_name: 'Alexander' })
    })
    it('responds with status 201', async () => {
        expect(res.status).toBe(201)
    })
    it('responds with da new (snapshot)', () => {
        expect(res.body).toMatchSnapshot()
        })
})

// describe('[POST] /api/students', () => {
//   it('responds with new student', async () => {
//       const res = await request(server)
//       .post('/api/students').send({ student_name: 'Alexander' })
//       expect(res.body).toMatchObject({ student_id: 5, student_name: 'Alexander' })
//   })
//   it('responds with status 201', async () => {
//       const res = await request(server)
//       .post('/api/students').send({ student_name: 'Alexander' })
//       expect(res.status).toBe(201)
//   })
// })


describe('[DELETE] /students/:id', () => {
    it('returns with a 200 Accepted status', async () => {
        const res = await request(server).delete('/api/students/1');
        expect(res.status).toBe(200);
    });
    it('deletes an item from the database', async () => {
        await request(server).delete('/api/students/2');
        const currentStudents = await db('students');
        expect(currentStudents).toHaveLength(3);
    });
    it('deletes the CORRECT item from the database', async () => {
        const res = await request(server).delete('/api/students/2');
        expect(res.body).toMatchObject({ student_id: 2, student_name: 'Bob' });
    });
});