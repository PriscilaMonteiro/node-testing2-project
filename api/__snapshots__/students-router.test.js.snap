// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POST] /api/students responds with da new (snapshot) 1`] = `
Object {
  "student_id": 5,
  "student_name": "Alexander",
}
`;
